ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/launcher/LaunchArgumentsHelper.java

```diff
@@ -240,44 +244,43 @@ public static Map<String, Object> getVMSpecificAttributesMap(ILaunchConfiguratio
 		return map;
 	}
 
-	public static String getTracingFileArgument(ILaunchConfiguration config, String optionsFileName) {
+	public static String getTracingFileArgument(ILaunchConfiguration config, Path optionsFile) {
 		try {
 			TracingOptionsManager mng = PDECore.getDefault().getTracingOptionsManager();
 			Map<String, String> options = config.getAttribute(IPDELauncherConstants.TRACING_OPTIONS, (Map<String, String>) null);
 			String selected = config.getAttribute(IPDELauncherConstants.TRACING_CHECKED, (String) null);
 			if (selected == null) {
-				mng.save(optionsFileName, options);
+				mng.save(optionsFile, options);
 			} else if (!selected.equals(IPDELauncherConstants.TRACING_NONE)) {
-				HashSet<String> result = new HashSet<>();
-				StringTokenizer tokenizer = new StringTokenizer(selected, ","); //$NON-NLS-1$
-				while (tokenizer.hasMoreTokens()) {
-					result.add(tokenizer.nextToken());
-				}
-				mng.save(optionsFileName, options, result);
+				Set<String> result = splitElementsByComma(selected).collect(Collectors.toSet());
+				mng.save(optionsFile, options, result);
 			}
 		} catch (CoreException e) {
 			return ""; //$NON-NLS-1$
 		}
-		return optionsFileName;
+		return optionsFile.toString();
 	}
 
 	public static String[] constructClasspath(ILaunchConfiguration configuration) throws CoreException {
 		double targetVersion = TargetPlatformHelper.getTargetVersion();
 		String jarPath = targetVersion >= 3.3 ? getEquinoxStartupPath(IPDEBuildConstants.BUNDLE_EQUINOX_LAUNCHER) : getStartupJarPath();
-		if (jarPath == null && targetVersion < 3.3)
+		if (jarPath == null && targetVersion < 3.3) {
 			jarPath = getEquinoxStartupPath("org.eclipse.core.launcher"); //$NON-NLS-1$
-
-		if (jarPath == null)
+		}
+		if (jarPath == null) {
 			return null;
-
-		ArrayList<String> entries = new ArrayList<>();
+		}
+		List<String> entries = new ArrayList<>();
 		entries.add(jarPath);
 
 		String bootstrap = configuration.getAttribute(IPDELauncherConstants.BOOTSTRAP_ENTRIES, ""); //$NON-NLS-1$
-		StringTokenizer tok = new StringTokenizer(getSubstitutedString(bootstrap), ","); //$NON-NLS-1$
-		while (tok.hasMoreTokens())
-			entries.add(tok.nextToken().trim());
-		return entries.toArray(new String[entries.size()]);
+		return splitElementsByComma(getSubstitutedString(bootstrap)).map(String::trim).toArray(String[]::new);
+	}
+
+	private static final Pattern COMMA = Pattern.compile(","); //$NON-NLS-1$
+
+	public static Stream<String> splitElementsByComma(String value) {
+		return COMMA.splitAsStream(value);
 	}
```

Comment:
I'd simplify it (remove `COMMA`) and I'd also trim the elements inside this method, just to make sure the white spaces don't cause any obscure trouble in the future:

```java
public static Stream<String> splitElementsByComma(String value) {
      return Arrays.stream(value.split(",")).map(String::trim); //$NON-NLS-1$
}
```

Reply:
Using trim directly is a good suggestion, but I'm not so convinced of using `value.split(",")` since this creates an array of all resulting elements of the split, while using `Pattern.splitAsStream()` creates a lazy stream of elements and therefore does not need the intermediate array and could terminate before reaching the end.
Of course it is definitely questionable if there is a real performance and memory usage gain from this small optimization (probably at least not a relevant one).